adoc/extensions/sycl_khr_command_graph.adoc

```diff
@@ -252,111 +253,423 @@ a@ Constructs a [code]#depends_on_all_leaves# property instance.
 
 |====
 
-=== Graph
+[[sec:khr-graph-command-graph]]
+== Command-graph class
+
+The [code]#command_graph# class represents a directed acyclic graph of nodes,
+where each node represents a single command with zero or more dependencies
+between them.
+For the list of supported node types see
+<<subsec:khr-graph-supported-node-types>>.
+
+A [code]#command_graph# is associated with a single SYCL device and context on
+which nodes representing device commands will be executed.
+The execution of a graph is considered complete when all of its nodes have
+completed.
+
+A [code]#command_graph# is built up by either recording queue submissions or
+explicitly adding nodes, then once the user is happy that the graph is complete,
+the graph instance is finalized into an executable variant - a fixed snapshot of
+the graph which can no longer have nodes added to it.
+Finalization may be a computationally expensive operation as the runtime may be
+able to perform optimizations based on the graph structure.
+However it can also perform work ahead of time that would normally be needed
+immediately prior to command execution.
+After finalization the graph can be submitted for execution on a queue one or
+more times with reduced launch overhead.
```

Comment:
This might be part of a broader conversation but this is an example of something I think we do a lot in the vendor extension which is describing usage in API definitions. I think the SYCL spec is a lot more segregated. For example in the architecture section they first describe something like the memory model with buffers and accessors, then the programming interface section for buffers is much more focused on the actual class and C++ usage.

To me this type of general description of what the extension does and how to use it belongs more in a general usage/description than in the class definition. Either this is just repeating what we've already said previously, or this is information hidden away in the API definitions that should be more easily available.